### DISCUSSION TITLE
How to focus on application main window after calling show

### DISCUSSION THREAD

**Discussion by frob** [beeware/toga#1683] [Q&A] [answered]
I am working to create an app with Toga that is able to be hidden with a key command and then shown with a key command regardless of if the program has focus. I was able to do this with a global keyboard shortcut and a background process. The issue that I am running into is focusing on the Toga app main_window after calling `show()` on it.
As it is I can `show` the app but if I am using another app I continue to use that other app. What I want to do is bring the Toga app to the front and use it with a key command. Another program that works this way is iterm.

**Comment by freakboy3742** [comment, marked as answer]
At present, there isn't an API for this. This sort of request is on the edge of what is the concern of an "app", and what is the concern of the operating system; while it's definitely something that *can* be done, it hasn't been an area of focus for Toga development.

**Reply by frob** [reply]
Would focusing on an element in the window be possible and the concern of the app? I am not sure if this would solve the problem. 

Would you suggest using a different gui framework in this case. tkinter or wxwidgets or something else.

**Reply by freakboy3742** [reply]
Giving a specific widget focus is definitely "in scope"; and for that, there's `.focus()`. I can't really comment on whether any other toolkit will be a better match for your use case.
